composer/callbacks/checkpoint_saver.py

```diff
@@ -319,6 +349,10 @@ def init(self, state: State, logger: Logger) -> None:
         folder = format_name_with_dist(self.folder, state.run_name)
         os.makedirs(folder, exist_ok=True)
 
+    def after_load(self, state: State, logger: Logger) -> None:
+        if not callable(self.save_interval):
+            self.save_interval = checkpoint_periodically(self.save_interval, state.timestamp)
+
```

Comment:
~Why are we moving this?~ Is this to ensure the starting timestamp is set? Can we add a comment if so?

Reply:
yeah, will add a comment